paf-mvp-core-js/tests/test-cases/ids-preferences-sign.test.ts

```diff
@@ -0,0 +1,43 @@
+import { Signer } from '@core/crypto/signer';
+import { PrivateKey } from '@core/crypto/keys';
+import { IdsAndPreferencesDefinition } from '@core/crypto/signing-definition';
+import { IdsAndPreferences } from '@core/model/generated-model';
+
+describe('IdsAndPreferencesDefinition', () => {
+  test('should sign ids and preferences', () => {
+    const mockPrivateKey: PrivateKey = {
+      sign: (data: string) => `SIGNED[${data}]`,
+    };
+
+    const idsAndPreferences: IdsAndPreferences = {
+      identifiers: [
+        {
+          version: '0.1',
+          type: 'paf_browser_id',
+          value: '7435313e-caee-4889-8ad7-0acd0114ae3c',
+          source: {
+            domain: 'operator.paf-operation-domain.io',
+            timestamp: 1642504380,
+            signature: 'id-signature',
+          },
+        },
+      ],
+      preferences: {
+        version: '0.1',
+        data: {
+          use_browsing_for_personalization: true,
+        },
+        source: {
+          domain: 'cmp.com',
+          timestamp: 1642504560,
+          signature: 'prefs-signature',
+        },
+      },
+    };
+
+    const signer = new Signer(mockPrivateKey, new IdsAndPreferencesDefinition());
```

Comment:
Looks like you are testing Signer instead of IdsAndPreferencesDefinition.
The main idea is to isolate classes from other codes and test them.
It means you should create an instance of IdsAndPreferencesDefinition and only test its public methods.
If your instance depends on other classes/implementations - they should be mocked or stabbed

Reply:
agreed and fixed